app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailScreen.kt

```diff
@@ -0,0 +1,230 @@
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Checkbox
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.android.architecture.blueprints.todoapp.R
+import com.example.android.architecture.blueprints.todoapp.data.Task
+import com.example.android.architecture.blueprints.todoapp.util.LoadingContent
+import com.example.android.architecture.blueprints.todoapp.util.getViewModelFactory
+import com.google.accompanist.appcompattheme.AppCompatTheme
+
+@Composable
+fun TaskDetailScreen(
+    taskId: String?,
+    onEditTask: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: TaskDetailViewModel = viewModel(factory = getViewModelFactory()),
+) {
+    val scaffoldState = rememberScaffoldState()
+    Scaffold(
+        scaffoldState = scaffoldState,
+        modifier = modifier.fillMaxSize(),
+        floatingActionButton = {
+            FloatingActionButton(onClick = onEditTask) {
+                Icon(Icons.Filled.Edit, stringResource(id = R.string.edit_task))
+            }
+        }
+    ) { paddingValues ->
+        val loading by viewModel.dataLoading.observeAsState(initial = false)
+        val dataAvailable by viewModel.isDataAvailable.observeAsState(initial = true)
+        val completed by viewModel.completed.observeAsState(initial = false)
+        val task by viewModel.task.observeAsState()
+
+        EditTaskContent(
+            loading = loading,
+            empty = !dataAvailable,
+            task = task,
+            taskCompleted = completed,
+            onRefresh = viewModel::refresh,
+            onTaskCheck = viewModel::setCompleted,
+            modifier = Modifier.padding(paddingValues)
+        )
+
+        // Note that the code below could be simplified with a state holder
+        // similar to AddEditTaskState
+
+        // Start loading data
+        LaunchedEffect(Unit) {
+            viewModel.start(taskId)
+        }
+
+        // Process snackbar events
+        val snackbarEvent by viewModel.snackbarText.observeAsState()
+        snackbarEvent?.getContentIfNotHandled()?.let { messageId ->
```

Comment:
Why is this not migrated to current guidance? A shared event? for migrations like these I'd like to see TODOs

Reply:
This is migrated in the last PR as well. I'm just migrating UI here, this actually works fine here, it starts to be an issue in the tasks screen.